### PR TITLE
Add `jax.nn.one_hot(x, num_classes, dtype)`.

### PR DESCRIPTION
**Pull request by tomhennigan**
This is an extremely common function and I've seen it re-implemented in a bunch of places, I think it would be useful to have a standard implementation in `jax.nn`.

I've marked `dtype` as keyword only because I think we may want to add more parameters to this function in the future (e.g. axis, on/off value) and we may want to place these args before dtype in the list (e.g. axis first). I'm not sure if there is an established pattern in JAX for this already and would be open to suggestions if there are alternatives to consider.

### CODE REVIEW

File: tests/nn_test.py [hunk 1]
```diff
@@ -61,6 +61,33 @@ def testHardTanhMemory(self):
     # see https://github.com/google/jax/pull/1640
     jax.make_jaxpr(nn.hard_tanh)(np.ones((10 ** 12,)))  # don't oom
 
+  def testOneHot(self):
+    actual = nn.one_hot(np.array([0, 1, 2]), 3)
```
Comment (gnecula) [inline tests/nn_test.py]: Maybe add a non-regular example, e.g., `one_hot([2, 0], 3)` ?

**Review by gnecula — APPROVED** [review]

File: tests/nn_test.py (hunk 1 above)
Comment (tomhennigan) [reply tests/nn_test.py]: Done, also fixed the f64 test case.

**Comment by mattjj** [comment]
Thanks, @tomhennigan !
